### ISSUE TITLE
Sudo check is brittle

### ISSUE THREAD

**Issue by MrMino** [ycm-core/ycmd#1593]
I just tried to reinstall my environment using my postinstall scripts. To my surprise, the installation failed on YCM compilation with:

```
This script should not be run with sudo.
```

The thing is, at the point of execution of the YCM compilation, I'm already past the point where the script dropped the privileges using `sudo -u $SUDO_USER`.

This check stems from b4cbf56962094c78f6e2b06a09c1375926144dee.

I don't think `sudo` by itself is an issue here, it looks like a very brittle attempt at checking for elevated privileges. Shouldn't this condition be `os.geteuid() == 0`?

**Comment by bstaletic** [comment]
Right, the idea is to check for elevated privileges, as those tend to mess with permissions and subsequently break things.

While I and @puremourning were discussing adding this check, I thought about `os.geteuid()`. The problem was that I didn't know the difference between `os.getuid()` and `os.geteuid()`. To be honest, I still don't know the difference.

@MrMino If you'd like, you can make a PR to improve the check.

**Comment by MrMino** [comment]
Thanks for the fast answer. Preparing PR as we speak.
